code/datums/components/limp.dm

```diff
@@ -0,0 +1,76 @@
+/*
+	Limping is a component to be attached to carbons for when they have a disparity in leg speed (such as when one of your legs is grievously wounded), so that one step has
+	a longer delay than the other. This is handled as a component rather than being inherent in bodypart code so that we're not constantly tracking footsteps if we don't need to
+
+	TODO: handle both legs being crippled
+	still have some edge cases i gotta figure out with removing this while having multiple limps
+*/
+
+/datum/component/limp
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+
+	var/obj/item/bodypart/l_leg/left
+	var/obj/item/bodypart/r_leg/right
+	var/obj/item/bodypart/next_leg
+	var/slowdown_left = 0
+	var/slowdown_right = 0
+	var/initialized
```

Comment:
The component is not very generic or expansible, and it looks like something that'd be better done with a trait that adds or removes movespeed modifiers, instead of checking every step.

Reply:
limp component is dead, replaced with limp status effect!